nx/source/runtime/devices/socket.c

```diff
@@ -584,9 +584,35 @@ int ioctl(int fd, int request, ...) {
     }
 }
 
+#define O_NONBLOCK_NX 0x800
+
+#define ALL_NX   (O_NONBLOCK_NX)
+#define ALL_FLAGS (O_NONBLOCK)
+
+static int from_nx(int flags) {
+    int newflags = 0;
```

Comment:
Where's O_NONBLOCK defined right now? 0x800 is what Linux uses I think, so this should already be correct? Can we fix this in the original header? If it's in newlib I guess it's not worth it to roll another version of it.

Follow codestyle so _FlagsFromNx()/_FlagsToNx()

Reply:
Current O_NONBLOCK value is wrong & is defined in newlib.

Reply:
newlib uses 0x4000 - https://github.com/devkitPro/newlib/blob/master/newlib/libc/include/sys/_default_fcntl.h

Interestingly linux has `# define O_NONBLOCK      04000`

I did think about changing stuff around in newlib but do we know all the flags yet? We translated for 3DS in libctru.